src/gov-action-contracts/AIPs/AIPXAction.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import {
+    ProxyAdmin,
+    TransparentUpgradeableProxy
+} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {
+    SecurityCouncilNomineeElectionGovernor,
+    GovernorSettingsUpgradeable
+} from "../../security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol";
+
+contract AIPXAction {
+    address public immutable proxyAdmin;
+    address public immutable nomineeElectionGovernorProxy;
+    address public immutable newNomineeElectionGovernorImplementation;
+    uint256 public immutable votingDelay;
+
+    constructor(
+        address _proxyAdmin,
+        address _nomineeElectionGovernorProxy,
+        address _newNomineeElectionGovernorImplementation,
+        uint256 _votingDelay
+    ) {
+        proxyAdmin = _proxyAdmin;
+        nomineeElectionGovernorProxy = _nomineeElectionGovernorProxy;
+        newNomineeElectionGovernorImplementation = _newNomineeElectionGovernorImplementation;
+        votingDelay = _votingDelay;
+    }
+
+    function perform() external {
+        // upgrade implementation
+        ProxyAdmin(proxyAdmin).upgrade(
```

Comment:
could also just use uprdadeAndCall

Reply:
let's also add a call to update the constitution text in here

Reply:
sounds good on constitution. i do think upgradeAndCall would be slightly less readable though.

Reply:
sounds good on the constitution update, upgradeAndCall seems to fail on owner check since the executor owns the governor

Reply:
6af99301530fdbbb70a50b028af4bcefa73a52f0

Reply:
> upgradeAndCall seems to fail on owner check since the executor owns the governor

ah true, nvrmind